fn.go

```diff
@@ -20,34 +34,200 @@ type Function struct {
 	log logging.Logger
 }
 
+const (
+	errFmtInvalidFunction   = "invalid function input: %s"
+	errFmtInvalidReadyValue = "%s is invalid, ready annotation must be True, Unspecified, or False"
+	errFmtInvalidMetaType   = "invalid meta kind %s"
+
+	errCannotGet   = "cannot get the function input"
+	errCannotParse = "cannot parse the provided templates"
+)
+
+const (
+	annotationKeyCompositionResourceName = "gotemplating.fn.crossplane.io/composition-resource-name"
+	annotationKeyReady                   = "gotemplating.fn.crossplane.io/ready"
+
+	metaApiVersion = "meta.gotemplating.fn.crossplane.io/v1alpha1"
+)
+
 // RunFunction runs the Function.
 func (f *Function) RunFunction(_ context.Context, req *fnv1beta1.RunFunctionRequest) (*fnv1beta1.RunFunctionResponse, error) {
 	f.log.Info("Running Function", "tag", req.GetMeta().GetTag())
 
-	// This creates a new response to the supplied request. Note that Functions
-	// are run in a pipeline! Other Functions may have run before this one. If
-	// they did, response.To will copy their desired state from req to rsp. Be
-	// sure to pass through any desired state your Function is not concerned
-	// with unmodified.
 	rsp := response.To(req, response.DefaultTTL)
 
-	// Input is supplied by the author of a Composition when they choose to run
-	// your Function. Input is arbitrary, except that it must be a KRM-like
-	// object. Supporting input is also optional - if you don't need to you can
-	// delete this, and delete the input directory.
 	in := &v1beta1.Input{}
```

Comment:
FWIW I was intending folks to rename `Input` to a kind that makes sense for the function - like maybe `Templates` here or something. Most people are not doing this so I should probably make it more obvious. 🙂